editor.planx.uk/src/pages/FlowEditor/components/Sidebar/EditHistory/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { gql, useSubscription } from "@apollo/client";
+import Box from "@mui/material/Box";
+import Divider from "@mui/material/Divider";
+import Typography from "@mui/material/Typography";
+import DelayedLoadingIndicator from "components/DelayedLoadingIndicator/DelayedLoadingIndicator";
+import { useStore } from "pages/FlowEditor/lib/store";
+import React from "react";
+import { Operation } from "types";
+import { AddCommentDialog } from "./AddCommentDialog";
+import { EditHistoryTimeline } from "./Timeline";
+
+export type HistoryItem = OperationHistoryItem | CommentHistoryItem | PublishHistoryItem;
+
+interface BaseHistoryItem {
+  id: number
+  createdAt: string;
+  actorId: number | undefined;
+  firstName: string;
+  lastName: string;
+}
+
+export interface OperationHistoryItem extends BaseHistoryItem {
+  type: "operation";
+  data: Operation["data"];
+  comment: null;
+}
+
+export interface CommentHistoryItem extends BaseHistoryItem {
+  type: "comment";
+  data: null;
+  comment: string;
+}
+
+export interface PublishHistoryItem extends BaseHistoryItem {
+  type: "publish";
+  data: null;
+  comment: string | null; // @todo require input in future publish modal changes
+}
+
+const EditHistory = () => {
+  const [flowId, canUserEditTeam, teamSlug, user] = useStore(
+    (state) => [
+      state.id,
+      state.canUserEditTeam,
+      state.teamSlug,
+      state.getUser(),
+    ],
+  );
+
+  const { data, loading, error } = useSubscription<{ history: HistoryItem[] }>(
+    gql`
+      subscription GetFlowHistory($flow_id: uuid = "") {
+        history: flow_history (
+          limit: 50
+          where: { flow_id: { _eq: $flow_id } }
+          order_by: { created_at: desc }
+        ) {
+          id
+          type
+          createdAt: created_at
+          actorId: actor_id
+          firstName: first_name
+          lastName: last_name
+          data(path: "op")
+          comment
+        }
+      }
+    `,
```

Comment:
I knew that GraphQL `subscription` only worked against a single database entity at a time (previously `operations` table), but I didn't know until this piece of work that we can do subscriptions against a _view_ ! 

It's a bit of a strange mental model for me, but works a charm for this scenario where the alternative would be three separate subscriptions to the `operations`, `flow_comments` and `published_flows` tables then having to concat & sort responses on the frontend !

Reply:
This is great to confirm! Much simpler than the alternative and allows us to combine and then sort a number of models here which is exactly what we're after.